vm/src/frame.rs

```diff
@@ -838,37 +843,35 @@ impl Frame {
 
     fn store_name(&self, vm: &mut VirtualMachine, name: &str) -> FrameResult {
         let obj = self.pop_value();
-        vm.ctx.set_attr(&self.locals, name, obj);
+        self.scope.locals.set_item(&vm.ctx, name, obj);
```

Comment:
We could implement the attributeprotocol for scope, what do you say? I feel a scope should have the option to set an attribute, not set an item.

Reply:
I thought about / started implementing DictProtocol. Maybe attribute protocol would make more sense.

It got a bit weird because you need to make decisions about who has responsibility for walking the parents scopes. Sometimes you want to just test a particular scope.

I've got a few more scope related things to do, so I'll look at this if it makes sense.